File: Toast Catalog/ImageRepository.m
#import "ImageRepository.h"

@implementation ImageRepository

- (UIImage *)imageForItemIdentifier:(NSInteger)itemIdentifier {
    NSString *systemImageName;
    if (itemIdentifier >= 50) {
        systemImageName = [NSString stringWithFormat:@"%li.circle.fill", (long)itemIdentifier];
        UIImage *image = [UIImage systemImageNamed:systemImageName];
        if (image) {
            return [image imageWithTintColor:[UIColor blackColor] renderingMode:UIImageRenderingModeAlwaysOriginal];
        }
    }
    return nil;
}



@end

File: Toast Catalog/ImageRepository.swift
//
//  ImageRep.swift
//  Toast Catalog
//
//  Created by Egor Anoshin on 23.04.2024.
//

import UIKit

class ImageRepository {
    func imageForItemIdentifier(_ itemIdentifier: Int) -> UIImage? {
        let systemImageName: String
        if itemIdentifier >= 50 {
            systemImageName = "\(itemIdentifier).circle.fill"
        } else {
            systemImageName = "questionmark.circle.fill"
        }
        return UIImage(systemName: systemImageName)
    }
}




